packages/eslint-plugin/tests/rules/no-explicit-any.test.ts

```diff
@@ -1199,8 +1199,68 @@ const test = <T extends Partial<never>>() => {};
         ],
         options: [{ ignoreRestArgs: true }],
       },
+      {
+        code: 'type Keys = keyof any;',
+        errors: [
+          {
+            column: 19,
+            line: 1,
+            messageId: 'unexpectedAny',
+            suggestions: [
+              {
+                messageId: 'suggestPropertyKey',
+                output: 'type Keys = PropertyKey;',
+              },
+            ],
+          },
+        ],
+      },
+      {
+        code: 'const integer = <TKey extends keyof any, TTarget extends { [K in TKey]: number }>(target: TTarget, key: TKey) => { /* ... */ };',
+        errors: [
+          {
+            column: 37,
+            line: 1,
+            messageId: 'unexpectedAny',
+            suggestions: [
+              {
+                messageId: 'suggestPropertyKey',
+                output:
+                  'const integer = <TKey extends PropertyKey, TTarget extends { [K in TKey]: number }>(target: TTarget, key: TKey) => { /* ... */ };',
+              },
+            ],
+          },
+        ],
+      },
     ] as RuleInvalidTestCase[]
   ).flatMap(testCase => {
+    if (testCase.code.includes('keyof any')) {
```

Comment:
[Testing] 😬 I know this is existing code with its own pattern, but - we're trying to move away from auto-generated tests. They're much harder to statically analyze and tend to batch together a lot of redundant tests with multiple edge cases per test. Could you please keep to manually written tests for the new code?

Reply:
Explicit suggestions for each is the way to go. If you search for `suggest`/`suggestions` in existing files you should be able to find examples.